animation_nodes/algorithms/mesh_generation/find_shortest_path.pyx

```diff
@@ -0,0 +1,126 @@
+from ... data_structures import GPStroke
+from ... data_structures cimport (
+    Mesh,
+    LongList,
+    ColorList,
+    FloatList,
+    PolySpline,
+    DoubleList,
+    Vector3DList,
+)
+
+from . line import getLinesMesh
+from ... math cimport distanceVec3
+
+# Dijkstra's algorithm (https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm) but is implemented
+# such a way to handle multi-sources and mesh with multiple islands.
+def getShortestPath(Mesh mesh, LongList sources, LongList targets, str pathType, str mode):
+    cdef Vector3DList vertices = mesh.vertices
+    cdef Py_ssize_t vertexCount = vertices.length
+    cdef float maxWeight = 1000000
+    cdef DoubleList weights = DoubleList.fromValue(maxWeight, length = vertexCount)
```

Comment:
Call this `distances`. Since our graph is unweighted.

Reply:
Done.